integrations/google_vertex/src/haystack_integrations/components/generators/google_vertex/chat/gemini.py

```diff
@@ -87,13 +93,25 @@ def __init__(
         self._model_name = model
         self._project_id = project_id
         self._location = location
-        self._model = GenerativeModel(self._model_name)
 
+        # model parameters
         self._generation_config = generation_config
         self._safety_settings = safety_settings
         self._tools = tools
+        self._tool_config = tool_config
+        self._system_instruction = system_instruction
         self._streaming_callback = streaming_callback
 
+        # except streaming_callback, all other model parameters can be passed during initialization
+        self._model = GenerativeModel(
+            self._model_name,
+            generation_config=self._generation_config,
+            safety_settings=self._safety_settings,
+            tools=self._tools,
+            tool_config=self._tool_config,
+            system_instruction=self._system_instruction,
```

Comment:
Initially, we were passing `tools` and other parameters directly in the `generate_content` method. However, these can be passed during model initialization instead, which simplifies parameter handling. See the reference [here](https://cloud.google.com/vertex-ai/generative-ai/docs/reference/python/latest/vertexai.preview.generative_models.ToolConfig). This approach simplifies the API's requirement that `system_instruction` can only be passed during model initialization.